### PR TITLE
Colin Dismuke - Spring-Challenge-RDBMS

### PR DESCRIPTION
**Pull request by cpdis**
🏃 @johnoro 🏃

### CODE REVIEW

**Comment by cpdis** [comment]
Sounds like I wasn't the only one having difficulty with showing the action and context arrays within the their projects and actions, respectively. I'll work on that over the next few days and hopefully get it implemented successfully.

File: data/migrations/20190112112210_schema_table.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      action.text("action_notes", 255).notNullable();
+      action.boolean("action_is_complete");
+      action.integer("projectId").unsigned();
+      action.foreign("projectId").references("projects.id");
+    });
+};
+
+exports.down = function(knex, Promise) {
+  // Rollback
+  return Promise.all(["projects", "actions"]).map(table =>
```
[the remark this answers is not in the document]
Comment (johnoro) [reply data/migrations/20190112112210_schema_table.js]: A better option might also be storing the table names from the get-go. Think something like this, as a simple example:
```javascript
const tables = [ 'projects', 'actions' ];
const [ projects, actions ] = tables;
// ...
  .createTable(projects, /* ... */ )
  .createTable(actions, /* ... */ )
// ...
  return tables.map( // ...
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+server.get("/", (req, res) => {
+  res.send(" ✅ Getting Things Done™️ ✅ ");
+});
+
+// Project endpoints
+server.get("/api/projects", (req, res) => {
+  helper
+    .getProjects()
+    .then(projects => {
+      res.status(200).json(projects);
```
Comment (johnoro) [inline index.js]: You _could_ skip the extra step and use `json` as your callback here:
```javascript
.then(res.status(200).json)
```

File: index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    })
+    .catch(err => {
+      res.status(500).json({ message: "Could not load projects." });
```
Comment (johnoro) [inline index.js]: Why pass `err` as a parameter when you're not using it? What might be preferred (if you don't have a good reason) is empty parentheses.

**Pull request closed without merging**